src/vm.rs

```diff
@@ -717,9 +720,11 @@ impl<'a, E: UserDefinedError, I: InstructionMeter> EbpfVm<'a, E, I> {
         }
 
         // Check config outside of the instruction loop
-        let instruction_meter_enabled = self.executable.get_config().enable_instruction_meter;
-        let instruction_tracing_enabled = self.executable.get_config().enable_instruction_tracing;
-        let dynamic_stack_frames = self.executable.get_config().dynamic_stack_frames;
+        let config = self.executable.get_config();
+        let instruction_meter_enabled = config.enable_instruction_meter;
```

Comment:
Are these indirections really necessary?
It feels like the rust compiler should be clever enough to optimize this loop invariant here.

Reply:
Oh I didn't do it with that in mind - I just did it since `config.foo` is easier on the eyes than `self.executable.get_config().foo` and we're only going to add more foos 😄